service/src/main/java/greencity/service/EventServiceImpl.java

```diff
@@ -601,6 +601,33 @@ private void addNewImages(Event toUpdate, UpdateEventDto updateEventDto, Multipa
         }
     }
 
+    private void validateEventRequest(AddEventDtoRequest addEventDtoRequest) {
+        checkingEqualityDateTimeInEventDateLocationDto(addEventDtoRequest.getDatesLocations());
+
+        if (!validateCoordinates(addEventDtoRequest.getDatesLocations())) {
+            throw new BadRequestException(ErrorMessage.INVALID_COORDINATES);
+        }
+        addAddressToLocation(addEventDtoRequest.getDatesLocations());
+    }
+
+    public boolean validateCoordinates(List<EventDateLocationDto> eventDateLocationDtos) {
+        for (EventDateLocationDto eventDateLocationDto : eventDateLocationDtos) {
+            AddressDto coordinates = eventDateLocationDto.getCoordinates();
+
+            if (coordinates == null || coordinates.getLatitude() == null || coordinates.getLongitude() == null) {
+                return false;
+            }
+
+            double latitude = coordinates.getLatitude();
+            double longitude = coordinates.getLongitude();
+
+            if (latitude < -90.0 || latitude > 90.0 || longitude < -180.0 || longitude > 180.0) {
```

Comment:
maybe move this check to separate method?

Reply:
Corrected